prowler/providers/aws/aws_provider.py

```diff
@@ -141,6 +140,36 @@ def generate_regional_clients(
         )
 
 
+def generate_regions_for_service(
```

Comment:
@Fennerr just a question before starting to review all the code in depth, why do we need to have both `generate_regional_clients` and `generate_regions_for_service`. Since the logic seems to be pretty much the same, could we merge this two functions into one?

Reply:
Hey, you are correct - this can be merged into 1

At first I wanted to generate a list of regions for a service without generating the regional clients, and then created boto3 sessions per region/thread, and then created the regional client from the session. But I did some more looking around online, and the boto clients are generally threadsafe (there are some caveats mentioned here: https://boto3.amazonaws.com/v1/documentation/api/1.19.0/guide/clients.html#multithreading-or-multiprocessing-with-clients), and I didn't like the long initialization time that it took each thread to setup it's own boto session, so I changed the logic.

Something else I am also concerned about is the thread_per_region always creating 1 thread per region. I am doing this off the assumption that the threads will be called ThreadPoolExecutorX_Y (X = the pool, Y = the threads, and it will increase from 0 to len(regions)). It seems to always work as intended so far, but I would welcome it if you have a more succinct way of ensuring that 1 thread per region is spawned.